// react-native-picasso/build/util/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PROPERTIES = exports.validPropertiesForType = exports.validPropertiesBase = exports.flexValues = exports.alignItemsValues = exports.alignValues = exports.borderWidthProperties = exports.spacingProperties = void 0;
exports.spacingProperties = [
    'm',
    'mr',
    'ml',
    'mt',
    'mb',
    'mx',
    'my',
    'p',
    'pr',
    'pl',
    'pt',
    'pb',
    'px',
    'py',
];
exports.borderWidthProperties = ['b', 'bl', 'br', 'bt', 'bb'];
exports.alignValues = {
    center: 'center',
    left: 'left',
    right: 'right',
    justify: 'justify',
};
exports.alignItemsValues = {
    center: 'center',
    start: 'flex-start',
    end: 'flex-end',
    stretch: 'stretch',
    baseline: 'baseline',
    between: 'space-between',
    around: 'space-around',
    evenly: 'space-evenly',
};
exports.flexValues = {
    row: 'row',
    column: 'column',
};
exports.validPropertiesBase = [...exports.spacingProperties, 'flex', 'alignself'];
const validPropertiesView = [
    'elevated',
    'radius',
    'radiustl',
    'radiustr',
    'radiusbl',
    'radiusbr',
    'bg',
    'alignitems',
    'justifycontent',
    'bordercolor',
    ...exports.borderWidthProperties,
];
const validPropertiesText = ['weight', 'align', 'color', 'size'];
exports.validPropertiesForType = {
    view: validPropertiesView,
    text: validPropertiesText,
    custom: [...validPropertiesView, ...validPropertiesText],
};
exports.PROPERTIES = {
    BACKGROUND: 'bg',
    COLOR: 'color',
    FONT_SIZE: 'size',
    FONT_WEIGHT: 'weight',
    ALIGN_ITEMS: 'alignitems',
    JUSTIFY_CONTENT: 'justifycontent',
    ALIGN_SELF: 'alignself',
    TEXT_ALIGN: 'align',
    FLEX: 'flex',
    BORDER_RADIUS: 'radius',
    BORDER_RADIUS_TOP_LEFT: 'radiustl',
    BORDER_RADIUS_TOP_RIGHT: 'radiustr',
    BORDER_RADIUS_BOTTOM_LEFT: 'radiusbl',
    BORDER_RADIUS_BOTTOM_RIGHT: 'radiusbr',
    BORDER_COLOR: 'bordercolor',
};
